### PR TITLE
CIS changes for new report format and config changes

### PR DESCRIPTION
**Pull request by leodotcloud**
Addresses:
https://github.com/rancher/rancher/issues/23904
https://github.com/rancher/rancher/issues/23905

### CODE REVIEW

File: pkg/controllers/user/cis/clusterScanHandler.go [hunk 1]
```diff
@@ -116,43 +143,44 @@ func (csh *cisScanHandler) Updated(cs *v3.ClusterScan) (runtime.Object, error) {
...
 
 	app := &projv3.App{
 		ObjectMeta: metav1.ObjectMeta{
 			Annotations: map[string]string{creatorIDAnno: creator.Name},
-			Name:        appName,
+			Name:        appInfo.appName,
 			Namespace:   appDeployProjectID,
 		},
 		Spec: projv3.AppSpec{
 			Answers:         appAnswers,
```
Comment (luthermonson) [inline pkg/controllers/user/cis/clusterScanHandler.go]: as long as passing "skip" will do what you want later then the rest of this is good. merge types and update go mod for new hashes then it's lgtm

File: pkg/controllers/user/cis/clusterScanHandler.go [hunk 2]
```diff
@@ -41,8 +51,30 @@ func (csh *cisScanHandler) Create(cs *v3.ClusterScan) (runtime.Object, error) {
...
+				appInfo.debugMaster = "true"
+			}
+			if cs.Spec.ScanConfig.CisScanConfig.DebugWorker {
+				appInfo.debugWorker = "true"
+			}
+		}
+		// Check if the configmap is populated
+		cm, err := csh.userCtxCMLister.Get(v3.DefaultNamespaceForCis, v3.ConfigMapNameForUserConfig)
+		if err != nil && !kerrors.IsNotFound(err) {
+			return cs, fmt.Errorf("error fetching cm: %v", err)
```
Comment (alena1108) [inline pkg/controllers/user/cis/clusterScanHandler.go]: can we say config map? :)

Comment (leodotcloud) [reply pkg/controllers/user/cis/clusterScanHandler.go]: `kubectl get cm` :) it's a well known acronym ... ha ha

Comment (leodotcloud) [reply pkg/controllers/user/cis/clusterScanHandler.go]: anyways, will change it :D

Comment (alena1108) [reply pkg/controllers/user/cis/clusterScanHandler.go]: on a second thought, we can just return the raw api error as it will be self explanatory

Comment (leodotcloud) [reply pkg/controllers/user/cis/clusterScanHandler.go]: Used full word and matched the format with the rest of the function.

**Comment by alena1108** [comment]
LGTM

**Comment by alena1108** [comment]
@leodotcloud was it built with go 1.13? I get a bunch of extra changes generated by `go mod vendor` on my machine, having 1.13.4

This pr can be merged only when vendor folder is updated with changes generated having go 1.13.4

**Comment by leodotcloud** [comment]
@alena1108 https://github.com/rancher/rancher/blob/master/Dockerfile.dapper#L20, if I use newer version of go (1.13.x), then drone wouldn't be happy.

**Comment by alena1108** [comment]
the pr is updated with go 1.13.4, so final LGTM

**Pull request merged**